.changeset/small-months-heal.md

```diff
@@ -0,0 +1,10 @@
+---
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Branch merges generate revertibles
+
+Merging any number of commits into a target branch (via the 'merge' method) now generates a revertible for each commit on the target branch.
```

Comment:
Does this include a revertible for the merge operation itself?

Reply:
Good question. There is no such thing as a merge commit, at least not as a primitive. Merge simply applies all the divergent changes of one branch to another branch (after rebasing them). The git-style "squash merge commit" is accomplished by the user running the merge inside of a transaction (because the transaction will squash all the commits that came in from the other branch). I expect that's going to be a pretty common pattern. Unfortunately we don't support it yet (you'll crash if you try) because there is some rebaser work that needs to be done first (I think it's in progress though!).